### ISSUE TITLE
Youtube-dl should normalize Unicode characters in filename to NFC on all platforms other than MacOS

### ISSUE THREAD

**Issue by madduck** [ytdl-org/youtube-dl#29056]
## Checklist

- [ ] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dl version **2021.04.26**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Verbose log

```
% youtube-dl -v https://soundcloud.com/magicianonduty/rammo-is-magician-on-duty
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://soundcloud.com/magicianonduty/rammo-is-magician-on-duty']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.04.26
[debug] Python version 3.9.2 (CPython) - Linux-5.10.0-6-amd64-x86_64-with-glibc2.31
[debug] exe versions: ffmpeg 4.3.2-0, ffprobe 4.3.2-0
[debug] Proxy map: {}
[soundcloud] magicianonduty/rammo-is-magician-on-duty: Downloading info JSON
[soundcloud] 320345934: Downloading JSON metadata
[soundcloud] 320345934: Downloading JSON metadata
[soundcloud] 320345934: Downloading JSON metadata
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on 'https://cf-media.sndcdn.com/q7MM70SDO4B5.128.mp3?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20vcTdNTTcwU0RPNEI1LjEyOC5tcDMiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE2MjExNTg0MzR9fX1dfQ__&Signature=SXai4OA0a0R2-qZuklTgPYgrHMpCtXWOGlKZOh6~UY5N8Fnxy3MT-4zGNmPbffijweNsU-3k3JhoJTMVMsxLs3ZJFdAf403SHuyaeeAqMGRGmOVJjpVvpZy2sDPFjqaL6pzamk6hQIJlMfnixuod-sgF~hfkLzpd2Wj9G~nj8texMQ2hs2493MCQGbXyu8~H7X5Dx67WzX4cKsJKexTdPTGjN8EpAJXOBJ~Tqytrbcdvj3~NR7TWVlTjJmvX2BFiTKTx4iyN0maDhB71ncToUsnhOvNZEiqEn8zX3NFy7ZvBlWhUYIJ~vjAP3dfH5DW~G-UWDYFLX64f8WoONW7xmQ__&Key-Pair-Id=APKAI6TU7MMXM5DG6EPQ'
[download] Destination: RAMMÖ is Magician on Duty-320345934.mp3
[download] 100% of 55.82MiB in 00:02

```

## Description

The name of the file created by `youtube-dl` contains a *combining character*, meaning that rather than the single "Ö" in the name (U+006d), it is actually made up of an "O" (U+004f), followed by a *combining diaeresis* "¨", which are then overlayed. Please refer to the [Syncthing documentation page on the subject](https://docs.syncthing.net/advanced/folder-autonormalize.html) for a good, short treatise of the subject. The [Wikipedia page on Unicode equivalence](https://en.wikipedia.org/wiki/Unicode_equivalence] goes into more detail. Unicode.org also has a [FAQ page on the matter](https://www.unicode.org/faq/normalization.html).

To avoid problems with multiple filenames that appear the same, but are made up of different bytes, operating systems have pretty much standardised on *not using* combining characters (called "NFC"), with MacOS being the only notable exception (using "NFD").

The issue here is that `youtube-dl` run on Linux creates files with combining characters in their filenames ("NFD"), and it should really be normalising these using "NFC" on all platforms, other than MacOS.

The [nfd2nfc project on PyPi](https://pypi.org/project/nfd2nfc/) may be useful, though there are surely other means to accomplish the same.

**Comment by madduck** [comment]
Youtube-dl is creating the file, and it's using a name that does not conform with the platform standards. How is this "out-of-scope"? Could you please provide a bit more information about your triage than just closing a bug I spent effort on to create according to your standards?

**Comment by eliteraspberries** [comment]
No need for an external package:

    filename = unicodedata.normalize('NFD', filename)

**Comment by dirkf** [comment]
This was implemented last year in c94a459a24:
```py
    # Replace look-alike Unicode glyphs
    if restricted and not is_id:
        s = unicodedata.normalize('NFKC', s)
```
Please open a new issue if this isn't working on your platform.
